internal/services/springcloud/spring_cloud_java_deployment_resource.go

```diff
@@ -50,11 +50,30 @@ func resourceSpringCloudJavaDeployment() *pluginsdk.Resource {
 				ValidateFunc: validate.SpringCloudAppID,
 			},
 
+			// TODO: Remove in 3.0
+			// The value returned in GET will be recalculated by the service if "cpu_v2" is honored, so make this property as Computed.
 			"cpu": {
-				Type:         pluginsdk.TypeInt,
-				Optional:     true,
-				Default:      1,
-				ValidateFunc: validation.IntBetween(1, 4),
+				Type:          pluginsdk.TypeInt,
+				Optional:      true,
+				Computed:      true,
+				ValidateFunc:  validation.IntBetween(1, 4),
+				ConflictsWith: []string{"cpu_v2"},
+				Deprecated:    "This field has been deprecated in favour of `cpu_v2` and will be removed in a future version of the provider",
+			},
+
+			// The value returned in GET will be recalculated by the service if "cpu" is honored, so make this property as Computed.
+			"cpu_v2": {
```

Comment:
what do these values mean? 

if it is 1-4ghz/500mhz instead of doing this change now how about we just remove this new property, and then allow 0 for cpu which is 0.5 mhz? (can we confirm that is the use?) and then in 3.0 change this to a float allowing 0.5
OR
we add a new property cpu_in_mhz and allow 500,1,2,3,4 and do this that way -

Reply:
These values are k8s notation of CPU units allocation. Since ASC is just AKS with managed spring under the hood - these notations tell you how many CPU units will be allocated to your container workload. 

Looking at 
https://github.com/Azure/azure-sdk-for-go/blob/eb6d24c94fabf2256e4a7818c83fae17ebb717b9/services/preview/appplatform/mgmt/2021-06-01-preview/appplatform/models.go#L2155
and
https://github.com/Azure/azure-sdk-for-go/blob/eb6d24c94fabf2256e4a7818c83fae17ebb717b9/services/preview/appplatform/mgmt/2021-06-01-preview/appplatform/models.go#L2784
considering that separate CPU and mem settings fields are deprecated - I'd introduce a new `resource_requests` block here looking like:
```
resource_requests {
  cpu = "500m"
  memory = "1GI" or "1024Mi" etc
}
```
then deprecate the other 2 fields and document the fact that the memory field just follows k8s notation, including validation of possible values in both "Gi" and "Mi" notation.

Reply:
These values are k8s notation of CPU units allocation. Since ASC is just AKS with managed spring under the hood - these notations tell you how many CPU units will be allocated to your container workload.

Looking at
https://github.com/Azure/azure-sdk-for-go/blob/eb6d24c94fabf2256e4a7818c83fae17ebb717b9/services/preview/appplatform/mgmt/2021-06-01-preview/appplatform/models.go#L2155
and
https://github.com/Azure/azure-sdk-for-go/blob/eb6d24c94fabf2256e4a7818c83fae17ebb717b9/services/preview/appplatform/mgmt/2021-06-01-preview/appplatform/models.go#L2784
considering that separate CPU and mem settings fields are deprecated - I'd suggest we introduce a new resource_requests block here looking like:

resource_requests {
  cpu = "500m"
  memory = "1Gi" or "1024Mi" etc
}
then deprecate the other 2 fields and document the fact that the memory field just follows k8s notation, including validation of possible values in both "Gi" and "Mi" notation.

Reply:
@mybayern1974 - after some internal discussion we've arrived at:
```
quota {
  cpu = "500m" or "1", "2", "4", "8"
  memory = "1Gi" or "1024Mi" etc
}
```

WDYT?